### PR TITLE
Improve fuzzing with recorded seed data

### LINKED ISSUE #2100: Improve fuzzing with recorded seed data
### What should be done?

`task test-unit` might be surprisingly slow due to a lot of files in the `tmp/records` directory accumulated by integration tests.

We should make the following improvements:
* store recorded files in nested directories and files matching their SHA-256 hashes without timestamps and random values (for example, given hash `abcdef[...]`, store file in `tmp/records/ab/abcdef[...].bin`);
* make `loadRecords` load up to random 1000 files.

### Where?

* https://github.com/FerretDB/FerretDB/blob/main/internal/clientconn/conn.go
* https://github.com/FerretDB/FerretDB/blob/main/internal/wire/records_test.go

### Definition of Done

- unit tests updated;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by fenogentov**
# Description

Closes #2100.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -277,7 +277,7 @@ tasks:
         --mode=diff-normal
         --handler=pg
         --postgresql-url=postgres://username@127.0.0.1:5432/ferretdb
-        --test-records-dir=tmp/records
+        --test-records-dir=./tmp/records
```
Comment (w84thesun) [inline Taskfile.yml:280]: Please revert those path changes. It will work as it is.

**Review by w84thesun — COMMENTED** [review]
I added a couple of comments. Please feel free to ping me on Slack if you need further help with this PR.

File: Taskfile.yml (hunk 1 above)
Comment (fenogentov) [reply Taskfile.yml:280]: This is temporary. My settings create files in an inconvenient place for debugging

File: internal/wire/records_test.go [hunk 2]
```diff
@@ -51,6 +53,15 @@ func loadRecords(recordsPath string) ([]testCase, error) {
...
 	}
 
+	// Select random N number of files from an array of files
+	N := 1000
+	if len(recordFiles) > N {
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		offset := r.Intn(len(recordFiles) - N)
+		limit := offset + N
+		recordFiles = recordFiles[offset:limit]
+	}
```
Comment (AlekSi) [inline internal/wire/records_test.go:63]: I think we could simplify that with https://pkg.go.dev/math/rand#Perm and use returned slice (up to 1000) as a list of indexes

Comment (fenogentov) [reply internal/wire/records_test.go:63]: but then we will always get the first 1000 files
It is acceptable?

**Review by fenogentov — COMMENTED** [review]
fix

**Review by fenogentov — COMMENTED** [review]
fix

File: internal/clientconn/conn.go [hunk 3]
```diff
@@ -203,16 +204,26 @@ func (c *conn) run(ctx context.Context) (err error) {
...
 
-			path := filepath.Join(
-				c.testRecordsDir,
-				fmt.Sprintf("%s-%d.bin", time.Now().Format("2006.01.02.15.04.05.000"), rand.Uint64()),
-			)
+			var hashPath string
+
+			hash := fmt.Sprintf("%64x", h.Sum(nil))
+			for i := 0; i < len(hash); i += 2 {
+				hashPath = filepath.Join(hashPath, hash[i:i+2])
```
[the remark this answers is not in the document]
Comment (fenogentov) [reply internal/clientconn/conn.go:211]: ambiguous interpretation of conditions) 
ok. will be fixed

[the remark this answers is not in the document]
Comment (AlekSi) [reply internal/clientconn/conn.go:211]: Actually, let's simplify that: let's use only one level and store the full hash inside. For example, given `abcdef0123…`, store it in `ab/abcdef0123…`. That matches how Go cache stores files, and is good enough, but simpler. I will update the issue.

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**